src/App.tsx

```diff
@@ -15,35 +16,108 @@
+  default = 'default',
+  alphabetically = 'alphabetically',
+  byLength = 'byLength',
+}
+
+interface SortParams {
+  sortOption: SortOption;
+  isReversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortOption, isReversed }: SortParams,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortOption) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortOption) {
+        case SortOption.alphabetically:
+          return good1.localeCompare(good2);
+        case SortOption.byLength:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortOption, setSortOption] = useState<SortOption>(SortOption.default);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const handleReset = () => {
+    setSortOption(SortOption.default);
+    setIsReversed(false);
+  };
+
+  const goods = getPreparedGoods(goodsFromServer, {
+    sortOption,
+    isReversed,
+  });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          onClick={() => setSortOption(SortOption.alphabetically)}
+          className={classNames('button is-info', {
+            'is-light': sortOption !== SortOption.alphabetically,
+          })}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          onClick={() => setSortOption(SortOption.byLength)}
+          className={classNames('button is-success', {
+            'is-light': sortOption !== SortOption.byLength,
+          })}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          onClick={() => setIsReversed(!isReversed)}
+          className={classNames('button is-warning', {
+            'is-light': !isReversed,
+          })}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortOption !== SortOption.default || isReversed) && (
+          <button
+            type="button"
+            onClick={() => handleReset()}
+            className="button is-danger is-light"
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {goods.map(good => (
+            <li key={good} data-cy="Good">
```

Comment:
Using the 'good' value as a key is not recommended if 'good' is not guaranteed to be unique. This can lead to rendering issues. Consider using a unique identifier for each item as a key, as suggested by the checklist item 2.